keras_contrib/applications/densenet.py

```diff
@@ -258,6 +370,38 @@ def __dense_block(x, nb_layers, nb_filter, growth_rate, bottleneck=False, dropou
     return x, nb_filter
 
 
+def __transition_up_block(ip, nb_filters, type='upsampling', output_shape=None, weight_decay=1E-4):
+    ''' SubpixelConvolutional Upscaling (factor = 2)
+
+    Args:
+        ip: keras tensor
+        nb_filters: number of layers
+        type: can be 'upsampling', 'subpixel', 'deconv', or 'atrous'. Determines type of upsampling performed
+        output_shape: required if type = 'deconv'. Output shape of tensor
+        weight_decay: weight decay factor
+
+    Returns: keras tensor, after applying upsampling operation.
+    '''
+
+    if type == 'upsampling':
+        x = UpSampling2D()(ip)
+    elif type == 'subpixel':
+        x = Convolution2D(nb_filters, 3, 3, activation="relu", border_mode='same', W_regularizer=l2(weight_decay),
+                          bias=False)(ip)
+        x = SubPixelUpscaling(r=2, channels=int(nb_filters // 4))(x)
+        x = Convolution2D(nb_filters, 3, 3, activation="relu", border_mode='same', W_regularizer=l2(weight_decay),
+                          bias=False)(x)
+    elif type == 'atrous':
+        # waiting on https://github.com/fchollet/keras/issues/4018
+        x = AtrousConvolution2D(nb_filters, 3, 3, activation="relu", W_regularizer=l2(weight_decay),
+                                bias=False, atrous_rate=(2, 2))(ip)
+    else:
+        x = Deconvolution2D(nb_filters, 3, 3, output_shape, activation='relu', border_mode='same',
+                            subsample=(2, 2))(ip)
+
+    return x
+
+
 def __create_dense_net(nb_classes, img_input, include_top, depth=40, nb_dense_block=3, growth_rate=12, nb_filter=-1,
```

Comment:
See comment above. The original densenet does not allow modification to this parameter, so i did not include it either.